addons/mail/static/src/messaging/component/composer/composer_tests.js

```diff
@@ -497,6 +492,284 @@ QUnit.test('composer text input cleared on message post', async function (assert
+                }, {
+                    email: 'demo@odoo.com',
+                    id: 7,
+                    name: 'Demo',
+                }],
+                name: "General",
+            }],
+        },
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'notify_typing') {
+                assert.step(`notify_typing:${args.kwargs.is_typing}`);
+                return;
+            }
+            return this._super(...arguments);
+        },
+        session,
+    });
+    const thread = this.env.entities.Thread.find(thread =>
+        thread.id === 20 &&
+        thread.model === 'mail.channel'
+    );
+    await this.createComposerComponent(thread.composer, { hasThreadTyping: true });
+
+    document.querySelector(`.o_ComposerTextInput_textarea`).focus();
+    document.execCommand('insertText', false, "a");
+    document.querySelector(`.o_ComposerTextInput_textarea`)
+        .dispatchEvent(new window.KeyboardEvent('keydown', { key: 'a' }));
+
+    assert.verifySteps(
+        ['notify_typing:true'],
+        "should have notified current partner typing status"
+    );
+});
+
+QUnit.test('current partner is typing should not translate on textual typing status', async function (assert) {
+    assert.expect(3);
+
+    const session = {
+        name: 'Admin',
+        partner_display_name: 'Your Company, Admin',
+        partner_id: 3,
+        uid: 2,
+    };
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 20,
+                members: [{
+                    email: 'admin@odoo.com',
+                    id: 3,
+                    name: 'Admin',
+                }, {
+                    email: 'demo@odoo.com',
+                    id: 7,
+                    name: 'Demo',
+                }],
+                name: "General",
+            }],
+        },
+    });
+    await this.start({
+        hasTimeControl: true,
+        async mockRPC(route, args) {
+            if (args.method === 'notify_typing') {
+                assert.step(`notify_typing:${args.kwargs.is_typing}`);
+                return;
+            }
+            return this._super(...arguments);
+        },
+        session,
+    });
+    const thread = this.env.entities.Thread.find(thread =>
+        thread.id === 20 &&
+        thread.model === 'mail.channel'
+    );
+    await this.createComposerComponent(thread.composer, { hasThreadTyping: true });
+
+    document.querySelector(`.o_ComposerTextInput_textarea`).focus();
+    document.execCommand('insertText', false, "a");
+    document.querySelector(`.o_ComposerTextInput_textarea`)
+        .dispatchEvent(new window.KeyboardEvent('keydown', { key: 'a' }));
+
+    assert.verifySteps(
+        ['notify_typing:true'],
+        "should have notified current partner typing status"
+    );
+
+    await nextAnimationFrame();
```

Comment:
no afterNextRender?

Reply:
because no re-render is triggered, purpose of this test is to check that current partner is not displayed in typing bar.

Reply:
I don't really understand the comment of the following line then.

> Thread textual typing status bar should display current partner is typing

Seems like the inverse of what you just said.